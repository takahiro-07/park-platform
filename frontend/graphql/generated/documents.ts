import * as Types from './types';

import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
export type CreateTagMutationVariables = Types.Exact<{
  input: Types.CreateTagInput;
}>;


export type CreateTagMutation = { __typename?: 'Mutation', createTag?: { __typename?: 'createTagPayload', tag: { __typename?: 'Tag', id: string, name: string, tagNumber: number, activeFlag: boolean } } | null };

export type DeleteTagMutationVariables = Types.Exact<{
  id: Types.Scalars['ID'];
}>;


export type DeleteTagMutation = { __typename?: 'Mutation', deleteTag?: { __typename?: 'deleteTagPayload', deleted: boolean } | null };

export type UpdateTagMutationVariables = Types.Exact<{
  input: Types.UpdateTagInput;
}>;


export type UpdateTagMutation = { __typename?: 'Mutation', updateTag?: { __typename?: 'updateTagPayload', tag: { __typename?: 'Tag', id: string, name: string, tagNumber: number, activeFlag: boolean } } | null };

export type TagQueryVariables = Types.Exact<{
  id: Types.Scalars['ID'];
}>;


export type TagQuery = { __typename?: 'Query', tag: { __typename?: 'Tag', id: string, name: string, tagNumber: number, activeFlag: boolean } };

export type TagsQueryVariables = Types.Exact<{ [key: string]: never; }>;


export type TagsQuery = { __typename?: 'Query', tags: Array<{ __typename?: 'Tag', id: string, name: string, tagNumber: number, activeFlag: boolean }> };


export const CreateTagDocument = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"mutation","name":{"kind":"Name","value":"CreateTag"},"variableDefinitions":[{"kind":"VariableDefinition","variable":{"kind":"Variable","name":{"kind":"Name","value":"input"}},"type":{"kind":"NonNullType","type":{"kind":"NamedType","name":{"kind":"Name","value":"createTagInput"}}}}],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"createTag"},"arguments":[{"kind":"Argument","name":{"kind":"Name","value":"input"},"value":{"kind":"Variable","name":{"kind":"Name","value":"input"}}}],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"tag"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"Field","name":{"kind":"Name","value":"tagNumber"}},{"kind":"Field","name":{"kind":"Name","value":"activeFlag"}}]}}]}}]}}]} as unknown as DocumentNode<CreateTagMutation, CreateTagMutationVariables>;
export const DeleteTagDocument = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"mutation","name":{"kind":"Name","value":"DeleteTag"},"variableDefinitions":[{"kind":"VariableDefinition","variable":{"kind":"Variable","name":{"kind":"Name","value":"id"}},"type":{"kind":"NonNullType","type":{"kind":"NamedType","name":{"kind":"Name","value":"ID"}}}}],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"deleteTag"},"arguments":[{"kind":"Argument","name":{"kind":"Name","value":"input"},"value":{"kind":"ObjectValue","fields":[{"kind":"ObjectField","name":{"kind":"Name","value":"id"},"value":{"kind":"Variable","name":{"kind":"Name","value":"id"}}}]}}],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"deleted"}}]}}]}}]} as unknown as DocumentNode<DeleteTagMutation, DeleteTagMutationVariables>;
export const UpdateTagDocument = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"mutation","name":{"kind":"Name","value":"UpdateTag"},"variableDefinitions":[{"kind":"VariableDefinition","variable":{"kind":"Variable","name":{"kind":"Name","value":"input"}},"type":{"kind":"NonNullType","type":{"kind":"NamedType","name":{"kind":"Name","value":"updateTagInput"}}}}],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"updateTag"},"arguments":[{"kind":"Argument","name":{"kind":"Name","value":"input"},"value":{"kind":"Variable","name":{"kind":"Name","value":"input"}}}],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"tag"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"Field","name":{"kind":"Name","value":"tagNumber"}},{"kind":"Field","name":{"kind":"Name","value":"activeFlag"}}]}}]}}]}}]} as unknown as DocumentNode<UpdateTagMutation, UpdateTagMutationVariables>;
export const TagDocument = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"query","name":{"kind":"Name","value":"Tag"},"variableDefinitions":[{"kind":"VariableDefinition","variable":{"kind":"Variable","name":{"kind":"Name","value":"id"}},"type":{"kind":"NonNullType","type":{"kind":"NamedType","name":{"kind":"Name","value":"ID"}}}}],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"tag"},"arguments":[{"kind":"Argument","name":{"kind":"Name","value":"id"},"value":{"kind":"Variable","name":{"kind":"Name","value":"id"}}}],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"Field","name":{"kind":"Name","value":"tagNumber"}},{"kind":"Field","name":{"kind":"Name","value":"activeFlag"}}]}}]}}]} as unknown as DocumentNode<TagQuery, TagQueryVariables>;
export const TagsDocument = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"query","name":{"kind":"Name","value":"Tags"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"tags"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"Field","name":{"kind":"Name","value":"name"}},{"kind":"Field","name":{"kind":"Name","value":"tagNumber"}},{"kind":"Field","name":{"kind":"Name","value":"activeFlag"}}]}}]}}]} as unknown as DocumentNode<TagsQuery, TagsQueryVariables>;
