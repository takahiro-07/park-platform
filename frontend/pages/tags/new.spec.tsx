import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { CreateTagDocument, CreateTagMutationVariables } from '../../graphql/generated/documents'
import { testRender } from '../../test/render'
import TagNewPage from './new'
import { graphql } from 'msw'

describe('Sampleコンポーネント', () => {
  const renderPage = testRender(<TagNewPage />)

  it('タグ作成', async () => {
    const mutationInterceptor = jest.fn()
    renderPage(
      graphql.mutation(CreateTagDocument, (req, res, ctx) => {
        mutationInterceptor(req.variables)
        return res.once(
          ctx.data({
            __typename: 'Mutation',
            createTag: {
              __typename: 'createTagPayload',
              tag: {
                __typename: 'Tag',
                id: "1",
                name: 'test1',
                tagNumber: 1,
                activeFlag: true,
              },
            },
          })
        )
      })
    )

    const inputTag = screen.getByPlaceholderText('タグ名')
    fireEvent.change(inputTag, { target: { value: 'test1' } })
    const inputTagNumber = screen.getByPlaceholderText('タグ番号')
    fireEvent.change(inputTagNumber, { target: { value: 1 } })
    const submitButton = screen.getByText('タグの追加')
    fireEvent.click(submitButton)

    expect(renderPage).toMatchSnapshot();

    await waitFor(() =>
      expect(mutationInterceptor).toHaveBeenCalledWith({
        input: {
          params: {
            name: 'test1',
            tagNumber: 1,
            activeFlag: true,
          },
        },
      } as CreateTagMutationVariables)
    )
  })
})
