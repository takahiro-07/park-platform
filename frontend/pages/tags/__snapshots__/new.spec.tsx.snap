// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sampleコンポーネント タグ作成 1`] = `[Function]`;
